index.d.ts

```diff
@@ -0,0 +1,36 @@
+import { MutableRefObject, PureComponent } from 'react';
+import { StyleProp, ViewStyle, ImageStyle } from "react-native";
+
+type AnimationConfig = {
+    [name: string]: number[]
+};
+
+type RequireOnlyOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> &
+    { [K in Keys]-?: Required<Pick<T, K>> & Partial<Record<Exclude<Keys, K>, undefined>> }[Keys];
+
+export interface ISpriteSheetProps {
+    source: number, // source must be required; { uri } will not work
+    columns: number,
+    rows: number,
+    ref: MutableRefObject<SpriteSheet | null>,
+    animations: AnimationConfig, // see example
+    viewStyle?: StyleProp<ViewStyle>, // styles for the sprite sheet container
+    imageStyle?: StyleProp<ImageStyle>, // styles for the sprite sheet
+    height?: number, // set either height, width, or none,
+    width?: number, // but not both height and width
+    onLoad?: () => void
+}
+
+export class SpriteSheet extends PureComponent<RequireOnlyOne<ISpriteSheetProps, "height" | "width">, {}> {
```

Comment:
It cause ts error as below when `width` and `height` are undefined. 

```
Property 'width' is missing in type '{ ref: RefObject<SpriteSheet>; source: number; columns: number; rows: number; animations: { basic: number[]; }; }' but required in type 'Readonly<Pick<SpriteSheetProps, "source" | "columns" | "rows" | "ref" | "animations" | "viewStyle" | "imageStyle" | "onLoad"> & Required<Pick<SpriteSheetProps, "width">> & Partial<...>>'.ts(2769)
```

`height` and `width` can be undefined both.

Reply:
I think `export default class SpriteSheet extends PureComponent<SpriteSheetProps, {}> {` is enough. It is not hard to find how to use `width` or `height` in documents(README).

Reply:
You'r right. I have fixed that. Thanks 👍